arcgis-ios-sdk-samples/Geometry/List transformations/ListTransformationsViewController.swift

```diff
@@ -0,0 +1,178 @@
+// Copyright 2018 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class ListTransformationsViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+
+    @IBOutlet var mapView: AGSMapView!
+    @IBOutlet var tableView: UITableView!
+    @IBOutlet var orderByMapExtent: UISwitch!
+    
+    var datatumTransformations: [AGSDatumTransformation] = []
+    var defaultTransformation: AGSDatumTransformation?
+    let graphicsOverlay = AGSGraphicsOverlay()
+    var originalGeometry = AGSPoint(x: 538985.355, y: 177329.516, spatialReference: AGSSpatialReference(wkid: 27700))
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["ListTransformationsViewController"]
+
+        // Get MapView from layout and set a map into this view
+        mapView.map = AGSMap(basemap: AGSBasemap.lightGrayCanvasVector())
+        mapView.graphicsOverlays.add(graphicsOverlay)
+        
+        //add original graphic to overlay
+        addGraphic(originalGeometry, color: .red, style: .square)
+        
+        mapView.map?.load(completion: { [weak self] (error) in
```

Comment:
Is there any advantage to doing that?

The autocomplete fills it out as above.

Reply:
The advantage is readability since the parameter label can be omitted. The way you have it isn't wrong, but In my experience it is more common to use a trailing closure than not.

I just learned something new. I was seeing autocomplete suggesting a trailing closure:

![screen shot 2018-07-17 at 1 52 50 pm](https://user-images.githubusercontent.com/2257493/42844754-cc6c81b8-89c8-11e8-8eb0-106102285934.png)

The difference seems to be the optional chaining. Since `load(completion:)` is being called on an optional map using optional chaining, autocomplete doesn't use a trailing closure. Fascinating!

Since you're creating the map in the same scope, I would suggest you create a variable for the map and call load from that rather than the map view's map property. That way you avoid having to unwrap the map.